app/models/category.rb

```diff
@@ -0,0 +1,10 @@
+class Category < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :category_payments
+  has_many :payments, through: :category_payments
+  has_one_attached :icon
+
```

Comment:
- [ ] Right now I am able to create a Category without a name or icon. But this should not be allowed since both are mandatory. Kindly add validation for the presence of `name` and `icon`.

Reply:
Fixed